Unpredictable fetching bytes instead of string

Sometimes when processing a stored procedure result set with multiple result sets, CHAR and VARCHAR columns will be returned as bytes instead of a string, even though the previous result set CHAR and VARCHAR columns were returned as strings correctly. This can be seen in this run of test_200_MultipleRsltsetsUniformColDefs.py:

```
======================================================================
FAIL: test_200_MultipleRsltsetsUniformColDefs (test_200_MultipleRsltsetsUniformColDefs.IbmDbTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_200_MultipleRsltsetsUniformColDefs.py", line 19, in test_200_MultipleRsltsetsUniformColDefs
    obj.assert_expect(self.run_test_200)
  File "/home/travis/build/kadler/python-ibmdb/IBM_DB/ibm_db/testfunctions.py", line 79, in assert_expect
    self.assertEqual(self.capture(testFuncName), self.expected_LUW(callstack[1][1]))
AssertionError: "Fetching first result setBubbles        [331 chars]ail)" != 'Fetching first result setBubbles        [310 chars]ail)'
- Fetching first result setBubbles         3Gizmo           4Peaches         1Pook            0Rickety Ride    5Smarty          2Sweater         6Fetching second result setb'Smarty          '2b'Pook            '0b'Peaches         '1b'Bubbles         '3Fetching third result setb'Bubbles         '3b'Gizmo           '4b'Pook            '0Fetching fourth result set (should fail)
?                                                                                                                                                                           --                - --                - ^^^^^^^^^^^^^^^^^^^ --                -                          --                - --                - --                -
+ Fetching first result setBubbles         3Gizmo           4Peaches         1Pook            0Rickety Ride    5Smarty          2Sweater         6Fetching second result setSmarty          2Pook            0Peaches         1Bubbles         3Fetching third result setBubbles         3Gizmo           4Pook            0Fetching fourth result set (should fail)
?                                                                                                                                                                                                             ^^^^^^^^^^^^^^^^
```

You can see that on the third result set (all 3 are from the animals table), the name column is return as bytes instead of strings and thus contain `b''` surrounding the data in the output.

--- a/IBM_DB/ibm_db/ibm_db.c
+++ b/IBM_DB/ibm_db/ibm_db.c
@@ -6687,6 +6687,7 @@ static PyObject *ibm_db_next_result(PyObject *self, PyObject *args)
 		new_stmt_res->s_bin_mode = stmt_res->s_bin_mode;
 		new_stmt_res->cursor_type = stmt_res->cursor_type;
 		new_stmt_res->s_case_mode = stmt_res->s_case_mode;
+		new_stmt_res->s_use_wchar = stmt_res->s_use_wchar;
 		new_stmt_res->head_cache_list = NULL;
 		new_stmt_res->current_node = NULL;
 		new_stmt_res->num_params = 0;

--- a/IBM_DB/ibm_db/ibm_db.c
+++ b/IBM_DB/ibm_db/ibm_db.c
@@ -6683,20 +6683,29 @@ static PyObject *ibm_db_next_result(PyObject *self, PyObject *args)
 
 		/* Initialize stmt resource members with default values. */
 		/* Parsing will update options if needed */
+		/* NOTE: This needs to match _ibm_db_new_stmt_struct */
 		new_stmt_res = PyObject_NEW(stmt_handle, &stmt_handleType);
+		new_stmt_res->hstmt = new_hstmt;
+		new_stmt_res->hdbc = stmt_res->hdbc;
+
 		new_stmt_res->s_bin_mode = stmt_res->s_bin_mode;
 		new_stmt_res->cursor_type = stmt_res->cursor_type;
 		new_stmt_res->s_case_mode = stmt_res->s_case_mode;
 		new_stmt_res->s_use_wchar = stmt_res->s_use_wchar;
+
 		new_stmt_res->head_cache_list = NULL;
 		new_stmt_res->current_node = NULL;
+
 		new_stmt_res->num_params = 0;
 		new_stmt_res->file_param = 0;
+
 		new_stmt_res->column_info = NULL;
 		new_stmt_res->num_columns = 0;
+
+		stmt_res->error_recno_tracker = 1;
+		stmt_res->errormsg_recno_tracker = 1;
+
 		new_stmt_res->row_data = NULL;
-		new_stmt_res->hstmt = new_hstmt;
-		new_stmt_res->hdbc = stmt_res->hdbc;
 
 		return (PyObject *)new_stmt_res;		
 	} else {

